storage/range.go

```diff
@@ -149,6 +152,7 @@ type Range struct {
 	Desc      *proto.RangeDescriptor
 	rm        RangeManager  // Makes some store methods available
 	splitting int32         // 1 if a split is underway; updated atomically
+	lastIndex uint64        // last index in the raft log; updated atomically
```

Comment:
Note in comment here that lastIndex is last written log index, not last applied log index. Will they ever be out of sync in reality, given that we process new log entries synchronously?